packages/taiko-client/driver/chain_syncer/blob/syncer.go

```diff
@@ -248,16 +248,41 @@ func (s *Syncer) onBlockProposed(
 		return fmt.Errorf("failed to fetch original TaikoL1.proposeBlock transaction: %w", err)
 	}
 
-	// Decode transactions list.
-	var txListFetcher txlistFetcher.TxListFetcher
-	if event.Meta.BlobUsed {
-		txListFetcher = txlistFetcher.NewBlobTxListFetcher(s.rpc.L1Beacon, s.blobDatasource)
-	} else {
-		txListFetcher = new(txlistFetcher.CalldataFetcher)
-	}
-	txListBytes, err := txListFetcher.Fetch(ctx, tx, &event.Meta)
+	// first see if we have a CalldataTxList event in the same transaction
+	// Get the transaction receipt
+	receipt, err := s.rpc.L1.TransactionReceipt(ctx, tx.Hash())
 	if err != nil {
-		return fmt.Errorf("failed to fetch tx list: %w", err)
+		return fmt.Errorf("failed to get transaction receipt: %w", err)
+	}
+
+	var txListBytes []byte
+	calldataTxListFound := false
+
+	// Iterate through the receipt logs
+	for _, vLog := range receipt.Logs {
+		// Use the generated binding's method to parse the log
+		event, err := s.rpc.TaikoL1.ParseCalldataTxList(*vLog)
+		if err == nil {
+			log.Info("CalldataTxList event found", "blockID", event.BlockId.String())
```

Comment:
Because now there can be multiple blocks being proposed in a single tx, we probably need to make sure that the block id here matches the one in the block proposed event we're handling?

Reply:
pushed up!